db/crud.go

```diff
@@ -630,6 +630,9 @@ func (db *Database) updateDoc(docid string, allowImport bool, expiry uint32, cal
 			return
 		}
 
+		// Test for shadower echo here before UpstreamRev gets set to CurrentRev
+		shadowerEcho = doc.CurrentRev == doc.UpstreamRev
```

Comment:
Unit test added creates new docs in shadow, updates a shadow doc with no body change (echo) and updates a shadow doc with body change.